pyccel/ast/core.py

```diff
@@ -2065,6 +2066,38 @@ def _ignore(cls, c):
         """
         return c is None or isinstance(c, (FunctionDef, *cls._ignored_types))
 
+class IndexedFunctionCall(FunctionCall):
+    """
+    Represents an indexed function call in the code.
+
+    Class representing indexed function calls, encapsulating all
+    relevant information for such calls within the code base.
+
+    Parameters
+    ----------
+    func : FunctionDef
+        The function being called.
+
+    args : tuple
+        The arguments passed to the function.
+
+    indexes : tuple
+        Indexes of kernel call.
+    """
+    __slots__ = ('_indexes',)
+    def __init__(self, func, args, indexes):
+        self._indexes = indexes
+        super().__init__(func, args, None)
```

Comment:
Or is this just because this class is only used in the syntactic stage? In that case maybe we should add an assert to make this explicit:
```python
assert pyccel_stage == "syntactic"
```

Reply:
[07cf232](https://github.com/pyccel/pyccel-cuda/pull/42/commits/07cf23209cb7d8ad74090fa3249cd5dd111f1193)